### DISCUSSION TITLE
[authentication]allow async valdidation

### DISCUSSION THREAD

**Discussion by lewoudar** [eadwinCode/django-ninja-extra#58] [Ideas]
Hi folks,
following #57 I'm wondering if we can't add asynchronous authentication support, but maybe this issue should be raised on django-ninja and not here 😅
Something like the following:

```python
class AuthBearer(AsyncHttpBearer):
    async def authenticate(self, request, token):
        # call an asynchronous api
        if token == "supersecret":
            return token
```

What do you think?

**Comment by eadwinCode** [comment]
What form of authentication support do you want to add?

**Reply by eadwinCode** [reply]
Or make asynchronous

**Comment by lewoudar** [comment]
I don't really have this use case, but one might want to connect to an [LDAP](https://en.wikipedia.org/wiki/Lightweight_Directory_Access_Protocol) server using an asynchronous call for example.
Nevertheless, I'm sure some users might need this asynchronous API.

**Reply by eadwinCode** [reply]
Does this not work for you with the Asyn Auth classes in NinjaExtra?

**Reply by lewoudar** [reply]
I don't know what Async Auth classes you are referring 😅

**Reply by eadwinCode** [reply]
I mean the Auth classes [here](https://github.com/eadwinCode/django-ninja-extra/blob/master/ninja_extra/security/api_key.py)

**Reply by lewoudar** [reply]
ooh! Nice! I wasn't aware of that, you need to document it for other users 😉

**Reply by eadwinCode** [reply]
I thought I did... I will do that shortly

**Reply by lewoudar** [reply]
👍
